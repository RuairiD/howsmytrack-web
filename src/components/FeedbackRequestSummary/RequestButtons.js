import React from "react";

import { Button, Popconfirm } from "antd";

const getDeleteConfirmationText = (mediaUrl) => {
    if (mediaUrl) {
        return "This request has not been assigned to a group. If you delete it, you will not receive any feedback on it. Are you sure you want to delete this request?";
    }
    return "This request has not been assigned to a group. Are you sure you want to delete this request?";
};

const RequestButtons = ({ mediaUrl, onEditClick, onDelete }) => (
    <React.Fragment>
        <Button
            shape="circle"
            icon="edit"
            onClick={onEditClick}
            style={{
                marginRight: "0.5em",
            }}
            className="request-button"
        />
        <Popconfirm
            title={getDeleteConfirmationText(mediaUrl)}
            onConfirm={onDelete}
            okText="Yes"
            cancelText="No"
        >
            <Button
                shape="circle"
                icon="delete"
                className="request-button"
            />
        </Popconfirm>
    </React.Fragment>
);

export default RequestButtons;
