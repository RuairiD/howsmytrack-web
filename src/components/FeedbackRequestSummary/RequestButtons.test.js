import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RequestButtons from "./RequestButtons";

describe("RequestButtons", () => {
    it("renders edit and delete buttons with a warning for deleting a request", () => {
        const wrapper = shallow(
            <RequestButtons
                mediaUrl="mediaUrl"
                onEditClick={() => {}}
                onDelete={() => {}}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders edit and delete buttons with a gentler warning for deleting a trackless request", () => {
        const wrapper = shallow(
            <RequestButtons
                onEditClick={() => {}}
                onDelete={() => {}}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
