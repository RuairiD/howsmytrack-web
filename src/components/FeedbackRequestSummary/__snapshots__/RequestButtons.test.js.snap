// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestButtons renders edit and delete buttons with a gentler warning for deleting a trackless request 1`] = `
<Fragment>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    icon="edit"
    loading={false}
    onClick={[Function]}
    shape="circle"
    style={
      Object {
        "marginRight": "0.5em",
      }
    }
  />
  <Popconfirm
    cancelText="No"
    disabled={false}
    icon={
      <Icon
        theme="filled"
        type="exclamation-circle"
      />
    }
    okText="Yes"
    okType="primary"
    onConfirm={[Function]}
    placement="top"
    title="This request has not been assigned to a group. Are you sure you want to delete this request?"
    transitionName="zoom-big"
    trigger="click"
  >
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      icon="delete"
      loading={false}
      shape="circle"
    />
  </Popconfirm>
</Fragment>
`;

exports[`RequestButtons renders edit and delete buttons with a warning for deleting a request 1`] = `
<Fragment>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    icon="edit"
    loading={false}
    onClick={[Function]}
    shape="circle"
    style={
      Object {
        "marginRight": "0.5em",
      }
    }
  />
  <Popconfirm
    cancelText="No"
    disabled={false}
    icon={
      <Icon
        theme="filled"
        type="exclamation-circle"
      />
    }
    okText="Yes"
    okType="primary"
    onConfirm={[Function]}
    placement="top"
    title="This request has not been assigned to a group. If you delete it, you will not receive any feedback on it. Are you sure you want to delete this request?"
    transitionName="zoom-big"
    trigger="click"
  >
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      icon="delete"
      loading={false}
      shape="circle"
    />
  </Popconfirm>
</Fragment>
`;
